ignite/cmd/plugin.go

```diff
@@ -71,6 +76,108 @@ func linkPluginCmds(rootCmd *cobra.Command, p *plugin.Plugin) {
 	}
 }
 
+func linkPluginHooks(rootCmd *cobra.Command, p *plugin.Plugin) {
+	if p.Error != nil {
+		return
+	}
+
+	for _, hook := range p.Interface.Hooks() {
+		linkPluginHook(rootCmd, p, hook)
+		if p.Error != nil {
+			return
+		}
+	}
+}
+
+func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook plugin.Hook) {
+	cmdPath := hook.PlaceHookOn
+
+	if !strings.HasPrefix(cmdPath, "ignite") {
+		// cmdPath must start with `ignite ` before comparison with
+		// cmd.CommandPath()
+		cmdPath = "ignite " + cmdPath
+	}
+
+	cmdPath = strings.TrimSpace(cmdPath)
+
+	cmd := findCommandByPath(rootCmd, cmdPath)
+
+	if cmd == nil {
+		p.Error = errors.Errorf("unable to find commandPath %q for plugin hook %q", cmdPath, hook.Name)
+		return
+	}
+
+	if !cmd.Runnable() {
+		p.Error = errors.Errorf("can't attach plugin hook %q to non executable command %s", hook.Name, hook.PlaceHookOn)
+		return
+	}
+
+	preRun := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		errors := make([]string, 0)
+		var err error
+		if preRun != nil {
+			err = preRun(cmd, args)
+			if err != nil {
+				// dont return the error, log it and let execution continue to `Run`
+				errors = append(errors, err.Error())
+			}
+		}
+
+		err = p.Interface.ExecuteHookPre(hook.Name, args)
+		if err != nil {
+			// dont return the error, log it and let execution continue to `Run`
+			errors = append(errors, err.Error())
+		}
+
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Are you sure you need this pause for the *pre* phase ? My assumption is the remaining execution should give enough time for the plugin to synchronize its I/O.

Reply:
I wanted to give time for response here as well as they are executed in separate frames of the stack. But if you don't think it's needed it can be removed

Reply:
Let's remove it then thank you.